### PR TITLE
chore: upgrade lru-cache; replace use of async-cache

### PR DESCRIPTION
**Pull request by trentm**
lru-cache@7 added support for async fetching, and async-cache
was deprecated

Refs: https://github.com/elastic/apm-agent-nodejs/issues/2760

---

Reviewer note: The changes in stacktraces.js is mostly about changing from callback-based async-cache usage to Promise-based lru-cache usage. Ignoring whitespace/indentation changes makes for a smaller diff: https://github.com/elastic/apm-agent-nodejs/pull/3610/files?w=1

### CODE REVIEW

**Review by david-luna — COMMENTED** [review]
all good, just need clarification on the `fetchMethod` of `sourceMapCache`

**Review by david-luna — APPROVED** [review]

File: lib/stacktraces.js [hunk 1]
```diff
@@ -28,43 +25,49 @@ function initStackTraceCollection() {
...
     }
-    loadSourceMap(filename, function onSourceMap(err, sourcemap) {
-      // Translate sourcemap===undefined to null, because 'async-cache'
-      // treats `undefined` as a cache miss. Without this there is no
-      // caching for files that have no sourcemap (the common case).
-      cb(err, sourcemap || null);
+    return new Promise((resolve, reject) => {
+      loadSourceMap(filename, function onSourceMap(err, sourcemap) {
+        if (signal.aborted) {
+          return;
```
[the remark this answers is not in the document]
Comment (trentm) [reply lib/stacktraces.js]: commit 8acc120e

File: lib/stacktraces.js [hunk 2]
```diff
@@ -305,139 +308,136 @@ function frameFromCallSite(
...
+      { filename, err: sourceMapErr },
+      'could not process file source map',
+    );
+  }
+  if (sourceMapConsumer) {
+    let pos;
+    try {
+      pos = sourceMapConsumer.originalPositionFor({
+        line: lineno,
+        column: callsite.getColumnNumber(),
```
[the remark this answers is not in the document]
Comment (trentm) [reply lib/stacktraces.js]: Done in commit 9feb5428

File: lib/stacktraces.js [hunk 3]
```diff
@@ -471,38 +471,49 @@ function gatherStackTrace(
...
-      );
+    // Fallback to parsing the err.stack string.
+    setImmediate(cb, null, stackTraceFromErrStackString(log, err));
+    return;
+  }
+
+  if (filterCallSite) {
+    callsites = callsites.filter(filterCallSite);
+    if (callsites.length === 0) {
+      // Note: I'm not sure using tthe fallback here is the intended behaviour,
```
[the remark this answers is not in the document]
Comment (trentm) [reply lib/stacktraces.js]: Done in commit 9feb5428

**Pull request merged**